### PR TITLE
Simplify type annotations `optuna/cli.py`

### PR DESCRIPTION
**Pull request by willdavidson05**
## Motivation

Work towards #4508  
## Description of the changes

- Added `from __future__ import annotations` to enable postponed evaluation of type annotations.
- Updated type hints from `List`, `Dict`, and `Tuple` to `list`, `dict`, and `tuple`.

### CODE REVIEW

**Review by nabenabe0928 — APPROVED** [review]
Thank you for the PR, LGTM

**Comment by c-bata** [comment]
@kAIto47802 Could you review this PR?

**Comment by c-bata** [comment]
@nabenabe0928 Could you please merge this PR after @kAIto47802 approves.

File: optuna/cli.py
Comment (kAIto47802) [inline optuna/cli.py:243]: An unnecessary blank line was inserted in your commit. Please consider removing it.
```suggestion
```

**Review by kAIto47802 — CHANGES_REQUESTED** [review]
Thank you for your PR! I left a minor comment.

**Comment by willdavidson05** [comment]
@kAIto47802 Thanks, should be good to go now!

**Pull request merged**
